db/models/comments.model.js

```diff
@@ -7,9 +8,25 @@ exports.fetchCommentsByReviewId = (fetchedComments) => {
       [fetchedComments.review_id]
     )
     .then((response) => {
-        if (!response.rows.length) {
-            return Promise.reject({ status: 404, message: "Path Not Found" });
-          }
+      if (!response.rows.length) {
+        return Promise.reject({ status: 404, message: "Path Not Found" });
+      }
       return response.rows;
     });
 };
+
+exports.createComment = (username, body, reviewId) => {
+  return db
+    .query(
+      `INSERT INTO comments (review_id, author, body)
+      VALUES ($1, $2, $3)
+      RETURNING*;`,
+      [reviewId, username, body]
+    )
+    .then((response) => {
+      return response.rows[0];
+    })
+    .catch((err) => {
```

Comment:
You already have a catch block in your controller that will catch any errors from your model. Remember that essentially you model is returning a Promise chain that we are accessing the fulfilment of in the controller. If whatever the model is returning were to reject, the catch block in the controller would catch this.

Reply:
I was just missing a 'next' somewhere so had added in all these extra bits to pass the tests!!